gateway/handlers/logs.go

```diff
@@ -0,0 +1,133 @@
+package handlers
+
+import (
+	"context"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+)
+
+const crlf = "\r\n"
+const upstreamLogsEndpoint = "/system/logs"
+
+// NewLogHandlerFunc creates and http HandlerFunc from the supplied log Requestor.
+func NewLogHandlerFunc(logProvider url.URL) http.HandlerFunc {
```

Comment:
@stefanprodan This is what code looks like when WriteTimeout is enabled on the http.Server.